src/lib/wallet/GoodWalletClass.js

```diff
@@ -594,9 +631,16 @@ export class GoodWallet {
    * @param {string} link
    * @returns {Promise<boolean>}
    */
-  isWithdrawLinkUsed(link: string): Promise<boolean> {
-    const { hasPayment } = this.oneTimePaymentsContract.methods
-    return hasPayment(link).call()
+  async isWithdrawLinkUsed(link: string): Promise<boolean> {
+    try {
+      const { hasPayment } = await this.oneTimePaymentsContract.methods
+      return hasPayment(link).call()
+    } catch (exception) {
+      const { message } = exception
+
+      log.warn('isWithdrawLinkUsed failed', message, exception)
+      throw exception
+    }
```

Comment:
getWithdrawDetails should be fixed too 

```
async getWithdrawDetails(otlCode: string): Promise<{ status: 'Completed' | 'Cancelled' | 'Pending' }> {
  try {
    const hashedCode = this.getWithdrawLink(otlCode)
    const { payments } = this.oneTimePaymentsContract.methods

    const { paymentAmount, hasPayment, paymentSender: sender } = 
      await this.oneTimePaymentsContract.methods.payments(hash).call()
    
    const amount = toBN(paymentAmount).toNumber()
    let status = WITHDRAW_STATUS_UNKNOWN

    // Check payment availability
    if (hasPayment && amount > 0) {
      status = WITHDRAW_STATUS_PENDING
    }

    if (hasPayment === false && toBN(paymentSender).isZero() === false) {
      status = WITHDRAW_STATUS_COMPLETE
    }

    return {
      hashedCode: hash,
      status,
      amount,
      sender: paymentSender,
    }
   }  catch (exception) {
       const { message } = exception

       log.warn('getWithdrawDetails failed', message, exception)
       throw exception
     }
  }
```

Reply:
Also, toBN **is sync function** so please adjust it's usage everywhere in `GoodWalletClass`